torch/quantization/observer.py

```diff
@@ -120,14 +126,80 @@ def calculate_qparams(self):
 
     @torch.jit.export
     def extra_repr(self):
-        return 'min_val={}, max_val={}'.format(self.min_val, self.max_val)
+        return "min_val={}, max_val={}".format(self.min_val, self.max_val)
+
+
+class HistogramObserver(ObserverBase):
+    r"""
+    The module records the running histogram of tensor values along with
+    min/max values. calculate_qparams will calculate scale and zero_point
+    """
+
+    __annotations__ = {
+        "min_val": Optional[torch.Tensor],
+        "max_val": Optional[torch.Tensor],
+        "relaxed_min": torch.Tensor,
+        "relaxed_max": torch.Tensor,
+        "histogram": Optional[torch.Tensor],
+    }
+
+    def __init__(self, bins=2048, **kwargs):
+        super(HistogramObserver, self).__init__(**kwargs)
+        self.bins = bins
+        self.histogram = None
+        self.min_val = None
+        self.max_val = None
+        self.relaxed_min = torch.tensor([0])
+        self.relaxed_max = torch.tensor([0])
+
+    def forward(self, x):
+        min_val = self.min_val
+        max_val = self.max_val
+        histogram = self.histogram
+        if min_val is None or max_val is None or histogram is None:
+            min_val = torch.min(x)
+            max_val = torch.max(x)
+            range = max_val - min_val
+            self.relaxed_min = min_val - 0.5 * range
+            self.relaxed_max = max_val + 0.5 * range
+            self.histogram = torch.histc(x, self.bins, min=self.relaxed_min, max=self.relaxed_max)
```

Comment:
It seems I trigged one issue for self.relaxed_min. I could put the convert(model) function under with torch.no_grad(): to work around the problem. 
`TypeError: histc(): argument 'min' must be Number, not Tensor`

Reply:
@llyfacebook, I've updated the PR to use self.relaxed_min.item() inside torch.histc(), which should resolve the issue. Could you let me know how to reproduce the TypeError so that I can double check?

Reply:
Just not put your convert(model) within "with torch.no_grad()", it should trigger the issue.

Reply:
I've tested the new PR works without error, thanks for your feedback @llyfacebook !